### ISSUE TITLE
[Android] System.NotSupportedException: Unable to activate instance of type myapp.Droid.Renderer.CustomTimePickerRenderer from native handle 0x7fe5975504 (key_handle 0x12ad2e0).

### ISSUE THREAD

**Issue by JulesMoorhouse** [xamarin/Xamarin.Forms#12044]
### Description
We're getting lots of crash reports from a CustomTimePickerRenderer on Android across a lot of devices. We couldn't  reproduce and trace the error, here is the error log from App Center

Android: 9
Android Build: PPR1.180610.011
Manufacturer: samsung
Model: SM-A530F

Xamarin Exception Stack:
System.NotSupportedException: Unable to activate instance of type myapp.Droid.Renderer.CustomTimePickerRenderer from native handle 0x7fe5975504 (key_handle 0x12ad2e0).
  at Java.Interop.TypeManager.CreateInstance (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer, System.Type targetType) [0x0017e] in <55654ebe9f2a48e6bade2862bb243f94>:0
  at Java.Lang.Object.GetObject (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer, System.Type type) [0x00023] in <55654ebe9f2a48e6bade2862bb243f94>:0
  at Java.Lang.Object._GetObject[T] (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer) [0x00017] in <55654ebe9f2a48e6bade2862bb243f94>:0
  at Java.Lang.Object.GetObject[T] (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer) [0x00000] in <55654ebe9f2a48e6bade2862bb243f94>:0
  at Java.Lang.Object.GetObject[T] (System.IntPtr jnienv, System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer) [0x00006] in <55654ebe9f2a48e6bade2862bb243f94>:0
  at Android.App.TimePickerDialog+IOnTimeSetListenerInvoker.n_OnTimeSet_Landroid_widget_TimePicker_II (System.IntPtr jnienv, System.IntPtr native__this, System.IntPtr native_view, System.Int32 hourOfDay, System.Int32 minute) [0x00000] in <55654ebe9f2a48e6bade2862bb243f94>:0
  at (wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.106(intptr,intptr,intptr,int,int)
--- End of inner exception stack trace ---
  System.MissingMethodException: No constructor found for myapp.Droid.Renderer.CustomTimePickerRenderer::.ctor(System.IntPtr, Android.Runtime.JniHandleOwnership)
    at Java.Interop.TypeManager.CreateProxy (System.Type type, System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer) [0x000b5] in <55654ebe9f2a48e6bade2862bb243f94>:0
    at Java.Interop.TypeManager.CreateInstance (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer, System.Type targetType) [0x00111] in <55654ebe9f2a48e6bade2862bb243f94>:0
  --- End of inner exception stack trace ---
    Java.Interop.JavaLocationException: Exception of type 'Java.Interop.JavaLocationException' was thrown.
      at Java.Lang.Error: Exception of type 'Java.Lang.Error' was thrown.
      at java.lang.Error: Java callstack:
      at crc643f46942d9dd1fff9.TimePickerRendererBase_1.n_onTimeSet(Native Method)
      at crc643f46942d9dd1fff9.TimePickerRendererBase_1.onTimeSet(TimePickerRendererBase_1.java:46)
      at android.app.TimePickerDialog.onClick(TimePickerDialog.java:172)
      at android.app.TimePickerDialog$1.onClick(TimePickerDialog.java:154)
      at android.view.View.performClick(View.java:7339)
      at android.widget.TextView.performClick(TextView.java:14222)
      at android.view.View.performClickInternal(View.java:7305)
      at android.view.View.access$3200(View.java:846)
      at android.view.View$PerformClick.run(View.java:27787)
      at android.os.Handler.handleCallback(Handler.java:873)
      at android.os.Handler.dispatchMessage(Handler.java:99)
      at android.os.Looper.loop(Looper.java:214)
      at android.app.ActivityThread.main(ActivityThread.java:7078)
      at java.lang.reflect.Method.invoke(Native Method)
      at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:964)

TypeManager.CreateProxy (System.Type type, System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer)
TypeManager.CreateInstance (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer, System.Type targetType)

TypeManager.CreateInstance (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer, System.Type targetType)
Object.GetObject (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer, System.Type type)
Object._GetObject[T] (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer)
Object.GetObject[T] (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer)
Object.GetObject[T] (System.IntPtr jnienv, System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer)
TimePickerDialog+IOnTimeSetListenerInvoker.n_OnTimeSet_Landroid_widget_TimePicker_II (System.IntPtr jnienv, System.IntPtr native__this, System.IntPtr native_view, System.Int32 hourOfDay, System.Int32 minute)
(wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.110(intptr,intptr,intptr,int,int)

Were using Xamarin Forms version 4.5.0.396

The error suggests that a constructor is missing... 
I've tried adding a constructor, but I'm not sure what base to use.
```
using Android.Content;
using myapp.Controls;
using myapp.Droid.Renderer;
using Xamarin.Forms;
using Xamarin.Forms.Platform.Android;

[assembly: ExportRenderer(typeof(CustomTimePicker), typeof(CustomTimePickerRenderer))]
namespace myapp.Droid.Renderer
{
	public class CustomTimePickerRenderer : TimePickerRenderer
	{
        public CustomTimePickerRenderer(Context context) : base(context)
        {
        }

        protected override void OnElementChanged(ElementChangedEventArgs<TimePicker> e)
	{
		base.OnElementChanged(e);

		if (Control != null)
		{
			Control?.SetBackgroundColor(Android.Graphics.Color.Transparent);
		}
	}
}
```

Here's some example xaml

```
        <control:CustomTimePicker x:Name="StartTimePicker"
                                  Time="{Binding StartTime}"
                                  HorizontalOptions="Start"
                                  TextColor="Black"/>

        <control:CustomTimePicker x:Name="EndTimePicker"
                                  Time="{Binding EndTime}"
                                  HorizontalOptions="Start"
                                  TextColor="{StaticResource Black}"/> 
```

Any help appreciated.

Thanks in advance.

**Comment by StephaneDelcroix** [comment]
are you sure you Preserve your type and methods from linking ?

**Comment by JulesMoorhouse** [comment]
We have `Don't Link` for the `Link behaviour` setting.. 

Following a chat, it sounds like we had a build error which suggested turning it to `Don't Link`.

I've tried  "Link SDK Assemblies only" and "Link all Assemblies"...

Link all... gives us an Xamarin.Form.Xaml.XamlParseException for a local boolToColorConverter...
Hmmm that's probably a genuine error :man_facepalming:

**Comment by PureWeen** [comment]
The exception indicates a memory leak or something is trying to access CustomTimePickerRenderer after it's been disposed of

I would override dispose on your CustomTimePickerRenderer to see when that's firing and see if you can correlate that to what's causing the exception

**Comment by Redth** [comment]
Since we haven't heard from you in more than 30 days, we hope this issue is no longer affecting you. If it is, please reopen this issue and provide the requested information so that we can look into it further. Thank you!

**Comment by zleao** [comment]
I've hit a similar problem and created a [sample repository](https://github.com/zleao/TimePickerActivationException) that shows an example on how to trigger this kind of exception.
In my case, I have an exception, when the TimePicker is being disposed when the TimePicker popup is being shown. I've checked the [Dispose method in the Android TimePickerRenderer](https://github.com/xamarin/Xamarin.Forms/blob/65609dca81382fbe445a78ebeea07563e57cbce1/Xamarin.Forms.Platform.Android/Renderers/TimePickerRenderer.cs#L125) and I noticed that the `_dialog` is not being hiden is visible.
In my case, forcing the [_dialog.Hide() before executing the dispose](https://github.com/zleao/TimePickerActivationException/blob/ab00da8a9dcf8564f0a8d26eea2d3da8dbfa4b34/TimePickerActivationExcpt/TimePickerActivationExcpt.Android/Renderers/MyTimePickerRenderer.cs#L44), solves the problem.
